packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,333 @@
+            )}
+        `
+    )}
+</div>
+```
+
+Header view:
+
+Note the following requires that `TableColumnMappingHeaderView` has a reference to the column with which it is associated. This is needed to enumerate the column's mapping elements.
+
+```HTML
+<div
+    ${ref('wrapper')}
+    class="wrapper"
+    @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.wrapper.innerText && x.wrapper.offsetWidth < x.wrapper.scrollWidth;
+    }}"
+    @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+    title=${x => (x.isValidContentAndHasOverflow ? x.wrapper.innerText : null)}
+>
+    ${repeat(x => (x.column as TableColumnMapping).mappings,
+        html<TableColumnMapping, TableColumnMappingHeaderView>`
+            ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+                html<TableColumnMapping>`
+                    ${repeat(x => x.mappedTemplate,
+                        html<HTMLTemplateElement>`${x => new ViewTemplate(x.innerHTML, [])}`
+                    )}
+                `
+            )}
+        `
+    )}
+</div>
+```
+
+For each mapping, if the key matches the value for this row, get the mapping's contained `template` and render its `innerHTML`.
+
+#### `nimble-mapping-string/number/boolean`
+
+```HTML
+<template slot="mapping">
+    <slot
+        ${slotted({ property: 'mappedTemplate', filter: x => x instanceof HTMLTemplateElement })}
+    >
+    </slot>
+</template>
+```
+
+### Grouping
+
+Will support grouping by the record value. The grouping header will display the rendered html mapped to that value. If multiple different values map to the same rendered html (e.g. icon), there may be separate groups that visually look like they should be combined into one group. We could instead render the record value into the header, but that value is not otherwise exposed to the user. It may not even have much meaning (e.g. some numeric code). I think it is more reasonable to display the rendered html.
+
+### Sorting
+
+Sorting has the same issue as grouping, where visually identical cells could end up not sorted contiguously. Again, I believe this is reasonable.
+
+### Sizing
+
+We will support fixed or fractional widths for this column type. We will introduce a new mixin for fixed-width support that exposes a `pixel-width` property. If this is set, the column will have a fixed width, otherwise it defaults to fractional width. There is an undocumented minimum column width, so a client may set `pixel-width` to 1 to end up with that minimum width.
+
+### Angular integration
+
+Angular directives will be created for the column component and the mapping components. The component will not have form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+A Blazor wrapper will be created for the component. _Initial attempts at prototyping resulted in exceptions being thrown and no icons being rendered. This is an open issue._
```

Comment:
For me this is a stronger indication that we need to question the API design being proposed. It's possible that the use of the `template` element is untenable in Blazor (for whatever reason), and if so, I don't think we could move forward with that as our design.

Reply:
+1  the template stuff is a new pattern not vetted yet, investigating sooner could prevent a lot of rework

Reply:
Figued out my issue. It seems Blazor elements aren't supported inside `template` elements. The result is that the `innerHTML`/`content` of the `HTMLTemplateElement` is empty. FAST chokes on the empty `ViewTemplate` I was creating, and that was the exception I was getting. Is it reasonable to direct Blazor clients to provide their mapped content using the Nimble elements rather than their Blazor wrappers?

Reply:
I think this is a reasonable suggestion. Particularly since we aren't expecting clients to bind aspects of the elements within the `template` to something on the parent component. They lose type safety for the attributes they wish to set on the components, but this may just be the cost of doing business in Blazor. Not sure if we can file an issue to ASP.NET Core asking for this to be fixed (this seems pretty corner case).

Reply:
Moot now with switch away from using `template` elements.